### PR TITLE
scalecube-services-benchmarks

### PR DESCRIPTION
**Pull request by segabriel**

### CODE REVIEW

**Comment by segabriel** [comment]
### How to launch
    java -cp scalecube-services-benchmarks-2.0.18-SNAPSHOT-jar-with-dependencies.jar .scalecube.services.benchmarks.RequestOneBenchmarksRunner

### How to launch and override params (with default params)
    java -cp scalecube-services-benchmarks-2.0.18-SNAPSHOT-jar-with-dependencies.jar .scalecube.services.benchmarks.RequestOneBenchmarksRunner nThreads=4 executionTaskTimeInSec=60 reporterPeriodInSec=10 responseCount=10

All extended params are optional

- nThreads, optional, number of used threads, Runtime.getRuntime().availableProcessors() by default.
- executionTaskTimeInSec, optional, how long execute each task in seconds, 20 sec by default.
- reporterPeriodInSec, how often reporter will print results, in seconds,  10 sec by default.
- responseCount, optional, how much responses we will wait for each request, 5 responses by default.

**Comment by ronenhamias** [comment]
i recommend to add a test that just send one request and specify how much to expect
like 'request(n)'

in such case its just one request and counting how much events return per sec.

File: benchmarks/pom.xml [hunk 1]
```diff
@@ -74,6 +80,30 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
```
Comment (artem-v) [inline benchmarks/pom.xml:83]: Try remove this assembly plugin and use just shade

File: rsocket-transport/src/main/java/io/scalecube/services/transport/rsocket/server/RSocketServiceAcceptor.java [hunk 2]
```diff
@@ -30,9 +30,23 @@ public RSocketServiceAcceptor(ServiceMessageHandler acceptor, ServiceMessageCode
...
+      public Flux<Payload> requestStream(Payload payload) {
+        return acceptor.requestStream(toMessage(payload))
+            .onErrorResume(t -> Flux.just(ExceptionProcessor.toMessage(t)))
+            .map(this::toPayload);
+      }
+
       @Override
       public Flux<Payload> requestChannel(Publisher<Payload> payloads) {
-        return acceptor.invoke(Flux.from(payloads).map(this::toMessage))
+        return acceptor.requestChannel(Flux.from(payloads).map(this::toMessage))
```
Comment (segabriel) [inline rsocket-transport/src/main/java/io/scalecube/services/transport/rsocket/server/RSocketServiceAcceptor.java:49]: 'toMessage' method can throw some exception, and we will not know about it. It needs to wrap into try-catch

**Review by ronenhamias — DISMISSED** [review]
void greeting test fails
tests are not running on travis so we cant merge it

**Comment by segabriel** [comment]
Benchmarks for RoundRobinServiceRouter

    Benchmark                                     (count)   Mode  Cnt        Score        Error  Units
    RouterBenchmarks.roundRobinRouterThroughput         1  thrpt    3  1065297.749 ± 740283.296  ops/s
    RouterBenchmarks.roundRobinRouterThroughput        10  thrpt    3   202341.851 ± 308850.328  ops/s
    RouterBenchmarks.roundRobinRouterThroughput       100  thrpt    3    22412.995 ±   1153.310  ops/s
    RouterBenchmarks.roundRobinRouterAverageTime        1   avgt    3     3927.878 ±    115.432  ns/op
    RouterBenchmarks.roundRobinRouterAverageTime       10   avgt    3    17477.033 ±    696.375  ns/op
    RouterBenchmarks.roundRobinRouterAverageTime      100   avgt    3   173403.971 ±   5737.451  ns/op

**Review by snripa — APPROVED** [review]

**Review by artem-v — APPROVED** [review]

**Pull request merged**